assets/src/data/eventespresso/core/model/relation-actions.js

```diff
@@ -0,0 +1,244 @@
+				entityId,
+				relationName,
+				addRelation
+			);
+		actions[ getMethodName(
+			modelName,
+			'relationsForIdAndRelationId',
+			'persist'
+		) ] = (
+			entityId,
+			relationName,
+			relationId,
+			addRelation = true
+		) => action
+			.persistRelationsForEntityIdAndRelationId(
+				modelName,
+				entityId,
+				relationName,
+				relationId,
+				addRelation
+			);
+		actions[ getMethodName(
+			modelName,
+			'id',
+			'removeDirtyRelationsFor'
+		) ] = (
+			entityId,
+			relationName,
+			relationId,
+			addRelation = true
+		) => action
+			.removeDirtyRelations(
+				relationName,
+				relationId,
+				modelName,
+				entityId,
+				addRelation
+			);
+		actions[ getMethodName(
+			modelName,
+			'id',
+			'receiveRelatedEntitiesFor'
+		) ] = ( entityId, relationName, relationEntityIds ) => action
+			.receiveRelatedEntities(
+				modelName,
+				entityId,
+				relationName,
+				relationEntityIds
+			);
+		actions[ getMethodName(
+			modelName,
+			'id',
+			'receiveDirtyRelationAdditionFor'
+		) ] = ( relationName, relationId, entityId ) => action
+			.receiveDirtyRelationAddition(
+				relationName,
+				relationId,
+				modelName,
+				entityId
+			);
+		actions[ getMethodName(
+			modelName,
+			'id',
+			'receiveDirtyRelationDeletionFor'
+		) ] = ( relationName, relationId, entityId ) => action
+			.receiveDirtyRelationDeletion(
+				relationName,
+				relationId,
+				modelName,
+				entityId
+			);
+		actions[ getMethodName(
+			modelName,
+			'relations',
+			'receiveUpdatedEntityIdFor'
+		) ] = ( oldId, newId ) => action
+			.receiveUpdatedEntityIdForRelations(
+				modelName,
+				oldId,
+				newId
+			);
+		actions[ getMethodName(
+			modelName,
+			'id',
+			'removeAllRelatedEntitiesFor'
+		) ] = ( entityId ) => action.removeAllRelatedEntitiesForModelEntity(
+			modelName,
+			entityId
+		);
+		actions[ getMethodName(
+			modelName,
+			'idAndRelation',
+			'removeRelatedEntitiesFor'
+		) ] = ( entityId, relationName, relationIds ) => action
+			.removeRelatedEntities(
+				modelName,
+				entityId,
+				relationName,
+				relationIds,
+			);
+		actions[ getMethodName(
+			modelName,
+			'idAndRelation',
+			'removeDirtyRelationIndexFor'
+		) ] = (
+			relationName,
+			relationEntityId,
+			entityId,
+			addRelation = true,
+		) => action.removeDirtyRelationIndex(
+			relationName,
+			relationEntityId,
+			modelName,
+			entityId,
+			addRelation
+		);
+		actions[ getMethodName(
+			modelName,
+			'idAndRelation',
+			'removeDirtyRelationAdditionFor'
+		) ] = (
+			relationName,
+			relationEntityId,
+			entityId,
+		) => action.removeDirtyRelationAddition(
+			relationName,
+			relationEntityId,
+			modelName,
+			entityId,
+		);
+		actions[ getMethodName(
+			modelName,
+			'idAndRelation',
+			'removeDirtyRelationDeletionFor'
+		) ] = (
+			relationName,
+			relationEntityId,
+			entityId,
+		) => action.removeDirtyRelationDeletion(
+			relationName,
+			relationEntityId,
+			modelName,
+			entityId,
```

Comment:
Nope, because this isn't calling `removeDirtyRelationForType`.  But as you pointed out earlier `removeDirtyRelationDeletion` does need to fix its internal code.